include/RAJA/policy/tensor/arch/hip/hip_wave.hpp

```diff
@@ -780,8 +780,8 @@ namespace expt
 
         // Third: mask off everything but output_segment
         //        this is because all output segments are valid at this point
-        // (5-segbits), the 5 is since the warp-width is 32 == 1<<5
-        int our_output_segment = get_lane()>>(6-segbits);
+        const int log2_warp_size = 32-1-__builtin_clz(warpSize);
```

Comment:
By using warpSize it appear that this is making a runtime value now instead of a compile time value, do we need to worry about the performance implications of this?

Reply:
It will add a few instructions and possibly a sgpr load. It probably won't be noticeable, but we can move the log2_warp_size as a constexpr member of the DeviceConstants. Or just wrap these two places in ifdefs. If we keep it dynamic then it should work for Nvidia/other as well - I'm not sure how important that is.

Reply:
Given that we're already relying on getting the right warpSize at configure time I think we should keep this compile time if possible. There is a constexpr RAJA::log2 that we can use instead of clz here to keep it constexpr.

Reply:
I finally had a chance to update this. I switched it over to the DeviceConstants/WARP_SIZE value using the RAJA::log2 call.